pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -256,12 +256,12 @@ func (r *Reconciler) exists() (bool, error) {
 	instance, err := r.computeService.InstancesGet(r.projectID, zone, r.machine.Name)
 	if err == nil {
 		switch instance.Status {
-		case "PROVISIONING", "REPAIRING", "RUNNING", "STAGING":
-			klog.Infof("Machine %q already exists", r.machine.Name)
-			return true, nil
-		default:
+		case "TERMINATED":
 			klog.Infof("Machine %q is considered as non existent as its status is %q", instance.Status)
```

Comment:
this belong to different commit

Reply:
This would break the delete behaviour. We want to keep re queuing until TERMINATED.

Reply:
yeh, I'd do first commit: "change exist to only false on terminated", second commit: "make delete async", not a blocker to me though.
Also we can drop the polling function now

Reply:
> yeh, I'd do first commit: "change exist to only false on terminated",

https://github.com/openshift/cluster-api-provider-gcp/pull/25

Reply:
> Also we can drop the polling function now

Done in 82a3f13ad1ae3c4c191e54957a226170c5bc63e2